packages/angular/cli/models/parser.ts

```diff
@@ -270,30 +284,73 @@ export function parseArguments(args: string[], options: Option[] | null): Argume
   const ignored: string[] = [];
   const errors: string[] = [];
 
-  for (let arg = args.shift(); arg !== undefined; arg = args.shift()) {
+  for (let i = 0; i < args.length; i++) {
```

Comment:
Instead of using 'i' which is already used below, use something similar to `argIndex`.   Eliminates the need to change the other loop to use `j`.